xbmc/interfaces/legacy/ListItem.cpp

```diff
@@ -169,6 +169,14 @@ namespace XBMCAddon
       item->GetVideoInfoTag()->SetRating(rating, votes, type, defaultt);
     }
 
+    void ListItem::AddSeason(int number, std::string name /* = "" */)
+    {
+      if (!item) return;
+
+      LOCKGUI;
```

Comment:
yeah, they certainly will. the locks are there because these callbacks were meant to modify items on screen (in a container), and in that case we can grab the gui lock. i do not recall all the gritty details, but the scrapers are called with another lock holding the thread at some point, resulting in deadlocks. i think it was when you run a 'manual' scan (i.e. update from info dialog, or 'i' on a unscanned item).